framework/wazuh/core/wazuh_queue.py

```diff
@@ -72,18 +57,55 @@ def _connect(self):
     def __enter__(self):
         return self
 
-    def _send(self, msg):
+    def _send(self, msg: bytes) -> None:
+        """Send a message through a socket.
+
+        Parameters
+        ----------
+        msg : bytes
+            The message to send.
+
+        Raises
+        ------
+        WazuhInternalError(1011)
+            If there was an error communicating with queue.
+        WazuhInternalError(1012)
+            If the size of the message is bigger than the buffer socket for Wazuh.
+        """
         try:
             sent = self.socket.send(msg)
 
             if sent == 0:
                 raise WazuhInternalError(1011, self.path)
-        except Exception:
+        except socket.error as e:
+            if e.errno == 90:
+                raise WazuhInternalError(1012, "The size of the event is longer than the buffer socket for Wazuh.")
```

Comment:
I've tested this and, despite what I said [here](https://github.com/wazuh/wazuh/pull/17023#discussion_r1218187732), it doesn't seem to be working right. For example, I am sending an event consisting of a list of numbers: "1 2 3 4 5 ... 26849 26850 26851". This is about 150000 bytes, but no socket.error is being thrown (although it does raise when sizes get even larger).

However in my tests, I can see that even though no error is produced, the last number I find in the archives.log is `12771` (about 65522 bytes). Therefore, part of the event has been discarded without warning the user, so this method is not effective for what we need (we should also open an issue to fix it in other integrations like AWS).

Please research how it is done in other parts of Wazuh or ask the core team if necessary about the most effective way to deal with this.